pkgs/development/tools/build-managers/mill/default.nix

```diff
@@ -0,0 +1,26 @@
+{ stdenv, lib, fetchurl, makeWrapper, jre, gnused }:
+ stdenv.mkDerivation rec {
+  name = "mill-${version}";
+  version = "0.3.5";
+    src = fetchurl {
```

Comment:
Use `fetchFromGitHub` here

Reply:
Thanks for all your feedback @alyssais ! 

Lihaoyi has prebuilt the Mill binary on this url:

* https://github.com/lihaoyi/mill/releases/download/0.3.5/0.3.5

I am not able to fetch that since it does not have the `.tar.gz` extension. 

I am not sure how easy it is to build this from source, since it is a build tool that is building itself. Would it be ok to keep the `fetchurl`. Are there any special requirements concerning using binaries like this in nixpkgs?

I've used sbt that is also a build tool in scala as my recipe:

https://github.com/scalavision/nixpkgs/blob/add-mill-build-tool/pkgs/development/tools/build-managers/sbt/default.nix

Reply:
Aha! I didn't catch that it was a binary. Generally builds from source are vastly preferred in Nixpkgs. Usually binaries are only used when source builds are impossible (eg if it's proprietary, closed-source software).

If it has a dependency on itself, something that, eg ghc and rustc do it to pull in a binary of the previous release, and use that to build the latest one from source.

Probably best to wait for somebody else to weigh in about the binary – personally I'd rather not have binaries in Nixpkgs when it's not absolutely necessary, but that's not my call.

Reply:
Yes.. I agree about binaries :-) I think creating a build using scalac would be quite some work though. I'll wait to see. Thank you very much for helping me out so far! :-)